pkg/apis/pipeline/v1alpha1/contexts.go

```diff
@@ -28,3 +28,25 @@ type hdcnKey struct{}
 func WithDefaultConfigurationName(ctx context.Context) context.Context {
 	return context.WithValue(ctx, hdcnKey{}, struct{}{})
 }
+
+// HasDefaultConfigurationName checks to see whether the given context has
+// been marked as having a default configurationName.
+func HasDefaultConfigurationName(ctx context.Context) bool {
+	return ctx.Value(hdcnKey{}) != nil
+}
+
+// lemonadeKey is used as the key for associating information
+// with a context.Context.
+type lemonadeKey struct{}
```

Comment:
what's lemonade?

Reply:
it's just a random key struct used internal for storing some metadata in the `context` (here just that we are upgrading it). I was debating on naming this sthg else, but it's how knative named it, and I like lemonade :stuck_out_tongue_closed_eyes:

Reply:
v good in hot weather 😎

Reply:
maybe the comment could mention that lemonadeKey is a random name? when i see this it makes me think there is some kinda important thing called "lemonade" - e.g. a "lemonade" defaulting algorithm?

Reply:
Ah I see, yeah I can add a comment :wink: